datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -229,7 +229,7 @@ def http(self):
     @property
     def metadata_manager(self):
         if self._metadata_manager is None:
-            if not self.check_id:
+            if not self.check_id and not using_stub_aggregator:
```

Comment:
Should we set a dummy `self.check_id` when using stub, instead of changing the code logic ?

Reply:
That is more intrusive imo

Reply:
IMO we should avoid conditionals that checks if we are in a test environment or not (here the stub).

The code behavior in test environment would be different than the real/prod environment. Which is something we want to avoid I think.

Reply:
We do it elsewhere when necessary https://github.com/DataDog/integrations-core/blob/334986696e8b104d1e718c0c3da15e2c51066f48/datadog_checks_base/datadog_checks/base/checks/base.py#L319-L326